sdk/common/src/main/java/io/opentelemetry/sdk/resources/detectors/ServiceDetector.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.resources.detectors;
+
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.sdk.resources.Entity;
+import io.opentelemetry.sdk.resources.EntityDetector;
+import java.util.Collections;
+import java.util.List;
+
+/** Detects the `service` entity. */
+public final class ServiceDetector implements EntityDetector {
+
+  private static final String SCHEMA_URL = "https://opentelemetry.io/schemas/1.28.0";
+  private static final String ENTITY_TYPE = "service";
+  private static final AttributeKey<String> SERVICE_NAME = AttributeKey.stringKey("service.name");
+
+  // Disabled as this would be break integration tests.
+  // private static final AttributeKey<String> SERVICE_NAMESPACE =
+  //     AttributeKey.stringKey("service.namespace");
+  //
+  // private static final AttributeKey<String> SERVICE_VERSION =
+  //     AttributeKey.stringKey("service.version");
+  //
+  //
+  // private static String getNamespace() {
+  //   // Fallback namespace is empty.
+  //   return "";
+  // }
+  //
+  // @Nullable
+  // private static String getVersion() {
+  //   // Specification lacks a way to specify non-fallback right now.
+  //   return null;
+  // }
+
+  private static String getServiceName() {
+    return System.getenv().getOrDefault("OTEL_SERVICE_NAME", "unknown_service:java");
```

Comment:
We separate out the interpretation of environment variables into autoconfigure, which has proved to be an separation of responsibilities. Should put all built-in entity detectors over there.

Reply:
You need to give me a solution to this then: https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/common/src/main/java/io/opentelemetry/sdk/resources/Resource.java#L51.  Effectively your "required" entity will be service from the SDK.  I can have the baked in discover a broken one, but then we need to make sure it appropriately gets overriden in all scenarios.

Additionally, be ready for: https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/common/src/main/java/io/opentelemetry/sdk/resources/Resource.java#L51. This *explicitly* not a configuration-based ENV variable.  This is a "carrier" of context for environments (e.g. K8S, FAAS, etc.) to provide identity to things they run. I think we should probably talk about how best to encode this in Java.  I'd rather not have this rely on autoconfigure, as that implies it's a configuration thing.